contracts/messaging/dfk/bridge/HeroBridge.sol

```diff
@@ -0,0 +1,199 @@
+import {HeroStatus} from "../types/HeroTypes.sol";
+
+pragma solidity 0.8.13;
+
+/** @title Core app for handling cross chain messaging passing to bridge Hero NFTs
+ */
+
+contract HeroBridge is SynMessagingReceiver {
+    address public heroes;
+    address public assistingAuction;
+    uint256 public msgGasLimit;
+
+    struct MessageFormat {
+        Hero dstHero;
+        address dstUser;
+        uint256 dstHeroId;
+    }
+
+    constructor(
+        address _messageBus,
+        address _heroes,
+        address _assistingAuction
+    ) {
+        messageBus = _messageBus;
+        heroes = _heroes;
+        assistingAuction = _assistingAuction;
+    }
+
+
+    /**
+     * @dev Whenever an {IERC721} `tokenId` token is transferred to this contract via {IERC721-safeTransferFrom}
+     * by `operator` from `from`, this function is called.
+     *
+     * It must return its Solidity selector to confirm the token transfer.
+     * If any other value is returned or the interface is not implemented by the recipient, the transfer will be reverted.
+     *
+     * The selector can be obtained in Solidity with `onERC721Received.selector`.
+     */
+    function onERC721Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        bytes calldata data
+    ) external returns (bytes4) { 
+        return this.onERC721Received.selector;
+    }
+
+    function _createMessage(
+        uint256 _heroId,
+        address _dstUserAddress,
+        Hero memory _heroToBridge
+    ) internal pure returns (bytes memory) {
+        // create the message here from the nested struct
+        MessageFormat memory msgFormat = MessageFormat({
+            dstHeroId: _heroId,
+            dstHero: _heroToBridge,
+            dstUser: _dstUserAddress
+        });
+        return abi.encode(msgFormat);
+    }
+
+    function _decodeMessage(bytes memory _message)
+        internal
+        pure
+        returns (MessageFormat memory)
+    {
+        MessageFormat memory decodedMessage = abi.decode(
+            _message,
+            (MessageFormat)
+        );
+        return decodedMessage;
+    }
+
+    function _createOptions() public returns (bytes memory) {
+        abi.encodePacked(uint16(1), msgGasLimit);
+    }
+
+    /**
+     * @notice User must have an existing hero minted to bridge it.
+     * @param _heroId specifics which hero msg.sender already holds and will transfer to the bridge contract
+     * @param _dstChainId The destination chain ID - typically, standard EVM chain ID, but differs on nonEVM chains
+     */
+    function sendHero(uint256 _heroId, uint256 _dstChainId) external payable {
+        Hero memory heroToBridge = IHeroCoreUpgradeable(heroes).getHero(
+            _heroId
+        );
+        bytes32 receiver = trustedRemoteLookup[_dstChainId];
+        // _createMessage(heroId, dstUserAddress, Hero);
+        bytes memory msgToPass = _createMessage(
+            _heroId,
+            msg.sender,
```

Comment:
Given that receiver of ERC721 is always `msg.sender` from source chain, we should specify that smart contracts aren't supposed to use `HeroBridge`, unless they're deployed on the same address on both chains (and support receiving ERC721)